Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -6541,9 +6541,13 @@ void WebPage::insertTextAsync(const String& text, const EditingRange& replacemen
         focusedElement->dispatchEvent(Event::create(eventNames().keydownEvent, Event::CanBubble::Yes, Event::IsCancelable::Yes));
 
     if (!frame->editor().hasComposition()) {
-        // An insertText: might be handled by other responders in the chain if we don't handle it.
-        // One example is space bar that results in scrolling down the page.
-        frame->editor().insertText(text, nullptr, replacesText ? TextEventInputAutocompletion : TextEventInputKeyboard);
+        if (text.isEmpty() && frame->selection().isRange())
```

Comment:
Should this logic also be implemented in `insertDictatedTextAsync`? Though this may not even be possible to do while dictating

Reply:
I don't think that's necessary, since `insertDictatedTextAsync` is only called if there are alternatives, and you can't have alternatives for empty text.